src/components/CreateContactSelector.tsx

```diff
@@ -0,0 +1,184 @@
+  };
+  return (
+    <div>
+      <div className="flex flex-row py-1">
+        <h2>Contacts</h2>
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <button
+              className="ml-auto rounded-lg bg-slate-200 p-2"
+              type="button"
+            >
+              add new contact
+            </button>
+          </DropdownMenuTrigger>
+          <DropdownMenuPortal>
+            <DropdownMenuContent>
+              <div className="mb-2 flex h-fit w-40 flex-wrap rounded-lg bg-slate-200 p-2 drop-shadow-md">
+                {available.map((plat) => (
+                  <DropdownMenuItem
+                    key={plat}
+                    onSelect={(e) => {
+                      addNew(plat);
+                    }}
+                  >
+                    <div className="group h-8 w-8" title={plat}>
+                      {logo[plat]}
+                    </div>
+                  </DropdownMenuItem>
+                ))}
+              </div>
+            </DropdownMenuContent>
+          </DropdownMenuPortal>
+        </DropdownMenu>
+      </div>
+      <div className="space-y-2">
+        {fields.map((field, index) => (
+          <ContactInput
+            key={field.id}
+            index={index}
+            register={register}
+            platform={field.platform}
+            remove={removeItem}
+            errors={errors}
+          />
+        ))}
+      </div>
+    </div>
+  );
+};
+export default ContactSelector;
+const styling = 'fill-black transition-colors group-hover:fill-blue-primary';
+const logo: logoProps = {
+  discord: <Discord className={styling} />,
+  youtube: <Youtube className={styling} />,
+  twitch: '/nebula-logo.png',
+  facebook: <Facebook className={styling} />,
+  twitter: <Twitter className={styling} />,
+  instagram: <Instagram className={styling} />,
+  website: <Website className={styling} />,
+  email: <Email className={styling} />,
+  other: '/Jupiter.png',
+};
+
+type ContactInputProps = {
+  register: UseFormRegister<z.infer<typeof createClubSchema>>;
+  remove: (index: number, platform: Contact['platform']) => void;
+  platform: Contact['platform'];
+  index: number;
+  errors: FieldErrors<z.infer<typeof createClubSchema>>;
+};
+const ContactInput = ({
+  register,
+  remove,
+  platform,
+  index,
+  errors,
+}: ContactInputProps) => {
+  return (
+    <div className="flex flex-row items-center bg-slate-200 p-2">
+      <div className="flex w-fit flex-row items-center rounded-md bg-slate-300 p-2">
+        <div className="box-content h-8 w-8">
+          <div className="h-8 w-8">{logo[platform]}</div>
+        </div>
+        <div className="text-xl">{platform}</div>
+      </div>
+      <div>
+        <div>Link here</div>
+        <input
+          type="text"
+          {...register(`contacts.${index}.url` as const)}
+          aria-invalid={
+            errors.contacts && errors.contacts[index]?.url ? 'true' : 'false'
```

Comment:
You can probably just leave this as `errors.contacts && errors.contacts[index]?.url` since it accepts boolean